content/blog/what_is_err_name_not_resolver_error_and_how_to_fix_It.md

```diff
@@ -0,0 +1,194 @@
+1. Open Chrome and go to Settings.
+2. Navigate to Privacy and Security.
+3. Disable the option "Preload pages for faster browsing and searching."
+
+Disabling this setting can stop {{< link href="https://www.google.com/chrome/dr/download/" >}}Chrome{{< /link >}} from attempting to pre-resolve websites that might conflict with DNS settings.
+
+{{< image "what_is_err_name_not_resolver_error_and_how_to_fix_It/perload_pages.webp" "Disable Browser Prediction Services" >}}
+
+#### 7. Check Your Firewall and Antivirus Settings
+In some cases, your firewall or antivirus software might block DNS queries, causing the error. Disabling your firewall or antivirus temporarily can help determine if they are the cause.
+
+**Solution**:
+- Temporarily disable your firewall or antivirus software.
+- If the website loads, adjust your firewall or antivirus settings to allow DNS queries, or whitelist the website you're trying to access.
+
+Make sure to enable your firewall or antivirus again after troubleshooting.
+
+#### 8. Verify Your Hosts File
+The hosts file on your computer contains mappings of IP addresses to domain names. If there are incorrect entries in this file, it could block access to certain websites.
+
+**Solution**:
+- **Windows**:
+  1. Open File Explorer and navigate to `C:\Windows\System32\drivers\etc\hosts`.
+  2. Open the file in a text editor (e.g., Notepad).
+  3. Remove or correct any incorrect entries.
+- **macOS**:
+  1. Open Terminal and type `sudo nano /etc/hosts`.
+  2. Review the file and remove any incorrect mappings.
+  3. Save the file and exit (press `CTRL + X`, then `Y` to confirm, and `Enter`).
+
+{{< image "what_is_err_name_not_resolver_error_and_how_to_fix_It/etc_host_terminal.webp" "Verify Your Hosts File" >}}
+
+Once the hosts file is corrected, try reloading the website.
+
+## How to Prevent ERR_NAME_NOT_RESOLVED Errors in the Future
+
+### Regularly Update Software and Drivers
+Keep your browser and network drivers updated to ensure compatibility and stability.
+
+### Use Reliable DNS Services
+Consider using reputable DNS providers like Google DNS (`8.8.8.8`, `8.8.4.4`) or Cloudflare DNS (`1.1.1.1`) for reliable performance and better security.
+
+### Periodically Clear DNS Cache
+Regularly clearing your DNS cache prevents outdated information from causing connectivity issues.
+
+### Maintain Proper Security Settings
+Regularly review your firewall and antivirus configurations to prevent accidental blocking of legitimate DNS queries.
+
+## Recommended Tools
+- **Cloudflare DNS (1.1.1.1)**: A fast, secure, and privacy-focused DNS resolution service.
+- **Google DNS (8.8.8.8, 8.8.4.4)**: Widely used and reliable DNS solution.
+- **DNS Jumper**: An easy-to-use tool that helps quickly switch between DNS providers, improving browsing speed and reliability.
+
+By following these comprehensive steps, you'll effectively resolve and prevent `ERR_NAME_NOT_RESOLVED` errors, ensuring smooth and uninterrupted internet browsing.
+
```

Comment:
At the end write one paragraph about "Fixing ERR_NAME_NOT_RESOLVED errors for Pinggy urls"

Mention that Pinggy assigns DNS dynamically when a tunnel is created. As a result, if the browser queries the url before the tunnel is setup, it may get ERR_NAME_NOT_RESOLVED. However the TTL of pinggy dns records is 10 seconds. So after around 10 to 15 seconds, refresh the page and it should work.